src/etos_test_runner/lib/executor.py

```diff
@@ -404,11 +405,21 @@ def _execute(self, workspace):
 
             self.logger.info("Wait for test to finish.")
             # We must consume the iterator here, even if we do not parse the lines.
-            for _, line in iterator:
+            proc = None
+            line = ""
+            for proc, line in iterator:
                 if self.test_regex:
                     self.parse(line)
             self.result = line
-            self.logger.info("Finished with result %r.", self.result)
+            if proc is not None:
+                self.returncode = proc.returncode
```

Comment:
Could we, instead of adding a new property, just return the exitcode from `_iterable_call` ?

Reply:
It would require modifying exit code from _iterable_call(), _execute() and execute(). Seems to be more reasonable to allow it to work same way as self.result.